File: CWM.Model/Models/WorkOrder.cs
using CWM.Core.Models.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CWM.Core.Models
{
    public class WorkOrder : Base
    {
        public string OrderNumber { get; set; } = string.Empty;
        public decimal Total {  get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public GarageBox GarageBox { get; set; }
        public Service ServicePerformed { get; set; }
        public string Concerne { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Sugestions { get; set; } = string.Empty;
        public Vehicle? Vehicle { get; set; } = new();
        public User? User { get; set; } = new();
        public Appointment? Appointment { get; set; } = new();
        public Employee? Employee { get; set; } = new();
    }
}

File: CWM.UI/cwm_desktop_mobile/lib/models/work_order.dart
import 'package:cwm_desktop_mobile/models/employee.dart';
import 'package:cwm_desktop_mobile/models/enums/garage_box.dart';
import 'package:cwm_desktop_mobile/models/enums/service.dart';
import 'package:cwm_desktop_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';
import 'appointment.dart';
import 'vehicle.dart';

part 'work_order.g.dart';

@JsonSerializable()
class WorkOrder {
  int id;
  String orderNumber;
  double total;
  DateTime startTime;
  DateTime endTime;
  GarageBox garageBox;
  Service servicePerformed;
  String concerne;
  String description;
  String sugestions;
  Vehicle? vehicle;
  User? user;
  Appointment? appointment;
  Employee? employee;

  WorkOrder(
      this.id,
      this.orderNumber,
      this.total,
      this.startTime,
      this.endTime,
      this.garageBox,
      this.servicePerformed,
      this.concerne,
      this.description,
      this.sugestions,
      this.vehicle,
      this.user,
      this.appointment);

  factory WorkOrder.fromJson(Map<String, dynamic> json) =>
      _$WorkOrderFromJson(json);

  Map<String, dynamic> toJson() => _$WorkOrderToJson(this);
}
